ONSdigital/florence
# New Auth Page

⚠️ Looks bigger than it actually is! ⚠️ 

### What

We are making a new Authentication user journey
This is the first PR of this journey which is the beginning of laying out the ground work

The idea is to keep the original Auth journeys and develop the New Auth journey in parallel. We will use a feature flag to toggle between the two different journeys. Once the entire Auth Epic work is complete we will delete the old login and auth work then remove the feature flags so that the new journeys are all that are left.

A decision has been made in the interim that if a change is minor and adds very little work we can make the changes without the feature flag and apply it to the current journey. An example of this is the Login screen title - which in this PR is applied to the new Sign In screens and the old login screen.

The terms:
- verb 'Log in'
- noun 'Login'
Will be replaced in upcoming work in one batch so for the time being from the users perspective they will see the term log in until the change of term work is added to the sprint.

The spirit of this Auth work is to keep PRs and tasks small. So rather than completing a whole page at a time, little changes will be introduced on each PR

#### So what is actually in this PR

- New feature flag used "ENABLE_NEW_SIGN_IN"
- 'Login' Florence page title replaced with "Log in to your Florence account"
- Line-height on the title increased
- Duplication of the loginController, LoginForm and LoginController.test - these will look more and more different in future PRs and tidied up a bit
- Tests (Config and LoginController) updated (but to be improved in future)

### How to review

1. Run Florence without the feature flag
2. Follow the login journey and ensure everything works like normal
3. The title of that login page should read "Log in to your Florence account"
4. Stop Florence and run: `make node-modules-react; make debug ENABLE_DATASET_IMPORT=true ENCRYPTION_DISABLED=true ENABLE_NEW_SIGN_IN=true;`
5. Navigate to the Florence login screen http://localhost:8081/florence/login
6. Ensure that the login screen title now says "Log in to your Florence account".

### Who can review

Anyone except me
